packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -270,17 +271,27 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 		return rest;
 	}
 
+	/**
+	 * Insert a new execution and its execution data using a transaction.
+	 */
 	async createNewExecution(execution: ExecutionPayload): Promise<string> {
-		const { data, workflowData, ...rest } = execution;
-		const { identifiers: inserted } = await this.insert(rest);
-		const { id: executionId } = inserted[0] as { id: string };
-		const { connections, nodes, name, settings } = workflowData ?? {};
-		await this.executionDataRepository.insert({
-			executionId,
-			workflowData: { connections, nodes, name, settings, id: workflowData.id },
-			data: stringify(data),
+		return await Db.transaction(async (transactionManager) => {
```

Comment:
can we please change `Db.transaction` to `this.manager.transaction`?
Neither is technically better than the other, but I'd like to get rid of the old `Db.ts` file at some point, and would prefer to not have more code tie into it.

Reply:
Yes, much better that way. I'll change